From 77346145fe1c5e9332b509ab489f087d29321c25 Mon Sep 17 00:00:00 2001
From: Guillaume Zajac <guillaume.zajac@linux.intel.com>
Date: Wed, 3 Oct 2012 10:18:00 +0200
Subject: [PATCH 04/10] mux: Declare new public API for HAL plugin
 manipulations
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Add new callback to notify when MUX is ready.
---
 include/mux.h |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/include/mux.h b/include/mux.h
index 1b11776..342244f 100644
--- a/include/mux.h
+++ b/include/mux.h
@@ -21,8 +21,15 @@
 #ifndef __MUX_H__
 #define __MUX_H__
 
-TReturn tcore_cmux_init(TcorePlugin *plugin, TcoreHal *hal);
-void    tcore_cmux_close(void);
-int     tcore_cmux_rcv_from_hal(unsigned char *data, size_t length);
+#include "hal.h"
+
+typedef void (*CMuxInitCallBack)(void *user_data);
+
+TReturn tcore_cmux_init(TcorePlugin *plugin, TcoreHal *hal,
+				CMuxInitCallBack cb, void *user_data,
+				int channel_num);
+TcoreHal *tcore_cmux_get_hal_channel(int channel_idx);
+void tcore_cmux_close(void);
+int tcore_cmux_rcv_from_hal(unsigned char *data, size_t length);
 
 #endif  /* __MUX_H__ */
-- 
1.7.10.4

